src/app/Http/Controllers/AccountController.php

```diff
@@ -348,4 +353,57 @@ public function updateMail(Request $request)
         }
         return redirect('/');
     }
+
+    public function pdfTicket(string $ticketId): Response {
+        $user = Auth::user();
+        $participant = EventParticipant::where('id', $ticketId)->first();
+        if ($user->id != $participant->user_id) {
+            return response()->view('errors.403', [], Response::HTTP_FORBIDDEN);
+        }
+
+        $event = Event::where('id', $participant->event_id)->first();
```

Comment:
you dont need to get the event here via ::where, you should use eloquent relations. that means, you can access it via $participant->event()

Reply:
Using relations